BraintreePSP/BraintreePaymentNonceProvider.swift

```diff
@@ -52,20 +52,29 @@ final class BraintreePaymentNonceProvider: PaymentNonceProvider {
             switch result {
             case .success(let sdkToken, _):
                 self?.sdkToken = sdkToken
+                self?.triggerAddCardFlow(
+                    token: sdkToken,
+                    currencyCode: quote.price.currencyCode,
+                    showUpdateCardAlert: false
+                )
             case .failure:
                 self?.callbackResult?(.completed(value: .failedToInitialisePaymentService(error: result.getErrorValue())))
                 return
             }
         }
-        triggerAddCardFlow(currencyCode: quote.price.currencyCode, showUpdateCardAlert: false)
     }
 
-    private func triggerAddCardFlow(currencyCode: String, showUpdateCardAlert: Bool) {
+    private func triggerAddCardFlow(
+        token: PaymentSDKToken?,
+        currencyCode: String,
+        showUpdateCardAlert: Bool
+    ) {
         self.cardRegistrationFlow.start(
             cardCurrency: currencyCode,
             amount: 0,
             supplierPartnerId: "",
             showUpdateCardAlert: showUpdateCardAlert,
+            dropInAuthenticationToken: token,
```

Comment:
woulnd't the `self.sdkToken` work here? You could remove one additional method parameter. If I understand correctly, every time code is calling this method, it's actually been using self.sdkToken.